modules/types/src/convert.ts

```diff
@@ -35,9 +35,19 @@ const getType = (input: any): NumericTypeName => {
 
 const castFunctions: any = {
   "bignumber-str": (x: BigNumber): string => x.toString(),
-  "number-bignumber": (x: number): BigNumber => new BigNumber(x),
+  "number-bignumber": (x: number): BigNumber => {
+    if (x < 0) {
+      throw new Error(`Input value ${x} cannot be a negative number`);
+    }
```

Comment:
we wouldnt want to check for this within the cast functions, those should be completely independent of validation

Reply:
I didn't like it either but I couldn't see how else to resolve https://github.com/ConnextProject/indra/issues/467 since conversion happens before validation. Should we do a pre-conversion check? Does it seem like we should at least try-catch around the bignumber cast?

Reply:
@LayneHaber any thoughts on this?

Reply:
We could, but the negative number shouldn't cause an issue when creating a new `BigNumber` instance, seems like the problem comes from the `_hex` property when it's an object?

We could check if it has a `_hex` field and then use the value of that property?

```sh
Laynes-MBP:cf-core laynehaber$ node
> const eth = require("ethers");
> new eth.utils.BigNumber(-1);
BigNumber { _hex: '-0x01' }
> new eth.utils.BigNumber("-0x01")
BigNumber { _hex: '-0x01' }
> new eth.utils.BigNumber({ _hex: '-0x01' })
{ Error: invalid BigNumber value (arg="value", value={"_hex":"-0x01"}, version=4.0.39)
    at Object.throwError (/Users/laynehaber/Developer/indra/node_modules/ethers/errors.js:76:17)
    at new BigNumber (/Users/laynehaber/Developer/indra/node_modules/ethers/utils/bignumber.js:99:20)
  reason: 'invalid BigNumber value',
  code: 'INVALID_ARGUMENT',
  arg: 'value',
  value: { _hex: '-0x01' } }
```

Reply:
Hmm okay well if it's not a negative number issue we should handle this differently. I don't see how this could be the _hex property though because all BigNumbers would have that. I'll look into this more.

Reply:
Found the issue, there was a missing validator and the validator had the wrong logic.